### PR TITLE
[glsl-in] Add support for gl_FragCoord / gl_FragDepth

### PR DESCRIPTION
**Pull request by magcius**
And remove the ability to read gl_Position in fragment.

### CODE REVIEW

File: src/front/glsl/variables.rs [hunk 1]
```diff
@@ -100,6 +100,25 @@ impl Program<'_> {
...
+                PrologueStage::FRAGMENT,
+            ),
+            "gl_FragDepth" => add_builtin(
+                TypeInner::Scalar {
+                    kind: ScalarKind::Float,
+                    width: 4,
+                },
+                BuiltIn::FragDepth,
+                true,
                 PrologueStage::FRAGMENT,
```
Comment (JCapucho) [inline src/front/glsl/variables.rs:122]: I don't think that it can be read in the fragment stage has its only defined as `out`

**Review by JCapucho — CHANGES_REQUESTED** [review]
Overall looks good, but I have some doubts regarding the `gl_FragDepth`

File: src/front/glsl/variables.rs (hunk 1 above)
Comment (magcius) [reply src/front/glsl/variables.rs:122]: Right, I forgot that PrologueStage was about reading. I wrote this before we chatted about that.

**Review by JCapucho — APPROVED** [review]
Looks good now

**Pull request merged**